lib/src/rules/library_private_types_in_public_api.dart

```diff
@@ -41,13 +41,22 @@ class _Private {}
 ''';
 
 class LibraryPrivateTypesInPublicAPI extends LintRule {
+  static const LintCode code = LintCode('library_private_types_in_public_api',
+      'Invalid use of a private type in a public API.',
+      correctionMessage:
+          'Try using a public supertype of the private type, or making the '
```

Comment:
I think the most common fix I've applied, in practice, is to actually change the definition of the type in the return type to be public. For example, a public method might return `List<_Foo>` and the `_Foo` is declared in the same library (obviously), and the fix is to rename `_Foo` to `Foo`.

Reply:
I agree, that's a better suggestion. I've updated the message.